src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs

```diff
@@ -284,7 +284,7 @@ private IReadOnlyList<ClientInfo> SetHierarchy(IReadOnlyDictionary<string, Clien
                 var clientNamespace = Configuration.Namespace;
                 var infoForEndpoint = topLevelClients.FirstOrDefault(c => c.ClientParameters.Any(p => p.IsEndpoint));
                 var endpointParameter = infoForEndpoint?.ClientParameters.FirstOrDefault(p => p.IsEndpoint);
-                var clientParameters = endpointParameter != null ? new[] { endpointParameter } : Array.Empty<InputParameter>();
+                var clientParameters = topLevelClients.SelectMany(c => c.ClientParameters.Where(p => !p.IsRequired || p.IsApiVersion || p.IsEndpoint)).Distinct().ToArray();
```

Comment:
What about other optional client paramters?

Reply:
Optional client parameter will be put in ClientOptions. So we still need optional client parameter, and may need to add int clientInfo.ClientParameters

Reply:
Normal topLevelClients will not go to this place. Only when user sets "single-top-level-client", we needing make up a new topLevelClient will go to this place. In this case, we need to bring the optional client parameters up to the new topLevelClient.

Reply:
Even it is a made-up topLevelClient when apply `single-top-level-client` we still need make sure this client contains all the needed parameters.